compiler/lib/generate.ml

```diff
@@ -208,15 +221,26 @@ module Share = struct
         else gen s
       with Not_found -> gen s
 
-  let get_apply gen n t =
+  let get_apply gen n exact tailcall t =
     if not t.alias_apply
-    then gen n
+    then gen n exact tailcall
     else
-      try J.EVar (IntMap.find n t.vars.applies)
+      try J.EVar (AppMap.find (n, exact, tailcall) t.vars.applies)
       with Not_found ->
-        let x = Var.fresh_n (Printf.sprintf "caml_call%d" n) in
+        let x =
+          Var.fresh_n
+            (Printf.sprintf
+               "caml_%scall%d"
+               (match exact, tailcall with
+               | true, false -> assert false
```

Comment:
Why assert false ?

Reply:
This case is handled directly in `translate_expr`.
https://github.com/ocsigen/js_of_ocaml/blob/a78b688979a5f9325e3be73f7c336be1bc3b849e/compiler/lib/generate.ml#L1075-L1077